### PR TITLE
fixes #829 - Add null string check and optional error message for apoc.spatial.geocode* functons

### PR DESCRIPTION
**Pull request by AngeloBusato**

### CODE REVIEW

File: src/main/java/apoc/spatial/Geocode.java [hunk 1]
```diff
@@ -245,4 +247,9 @@ public GeoCodeResult(Double latitude, Double longitude, String description, Map<
             this.location = map("latitude", latitude, "longitude", longitude, "description", description);
         }
     }
-}
+
+    public void checkNullOrEmptyAddress(String address){
```
Comment (inserpio) [inline src/main/java/apoc/spatial/Geocode.java]: static

File: src/test/java/apoc/spatial/SpatialTest.java [hunk 2]
```diff
@@ -41,19 +38,26 @@ public MockGeocode() {
...
+            checkNullOrEmptyAddress(address);
             if (geocodeResults != null && geocodeResults.containsKey(address)) {
                 Map data = geocodeResults.get(address);
                 return Stream.of(new Geocode.GeoCodeResult(Util.toDouble(data.get("lat")), Util.toDouble(data.get("lon")), String.valueOf(data.get("display_name")), data));
             } else {
                 return Stream.empty();
             }
         }
+
+        public void checkNullOrEmptyAddress(String address){
```
Comment (inserpio) [inline src/test/java/apoc/spatial/SpatialTest.java]: static

Comment (inserpio) [reply src/test/java/apoc/spatial/SpatialTest.java]: or call `Geocode. checkNullOrEmptyAddress `

**Review by inserpio — CHANGES_REQUESTED** [review]

File: src/main/java/apoc/spatial/Geocode.java [hunk 3]
```diff
@@ -245,4 +247,9 @@ public GeoCodeResult(Double latitude, Double longitude, String description, Map<
             this.location = map("latitude", latitude, "longitude", longitude, "description", description);
         }
     }
-}
+
+    public static void checkNullOrEmptyAddress(String address){
+        if (address == null || address.isEmpty() || address.equals(""))
```
Comment (albertodelazzari) [inline src/main/java/apoc/spatial/Geocode.java]: address.isEmpty() == address.equals("") because "".length() == 0

**Review by albertodelazzari — APPROVED** [review]

**Pull request merged**